File: Starry_Night.py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap

# Load the image of Starry Night!
image_path = 'testt.jpg'  # Replace with the actual path to your image
img = plt.imread(image_path)

# Extract the red channel (you can modify this based on the colors you want)
red_channel = img[:, :, 0]

# Create a heatmap using the red channel values
cmap = plt.cm.jet  # You can use other colormaps as well
heatmap = plt.imshow(red_channel, cmap=cmap)

# Add a colorbar legend
cbar = plt.colorbar(heatmap)
cbar.set_label('Intensity', rotation=270, labelpad=15)

# Add title and show the plot
plt.title('Wheat Field with Cypresses - Van Gogh Heatmap')
plt.show()

File: Starry_Night.m
clear,clc
% Load the image of Starry Night.
img = imread("C:\Users\Sina\Desktop\starry_night.jpg"); % Replace with the actual path to your image

% % Convert the image to grayscale
% grayscale_img = rgb2gray(img);
% 
% % Create a heatmap using the grayscale values
% colormap_custom = [0, 0, 0; 0, 0, 1; 0.5, 0, 0.5; 0, 0, 0.5; 0, 0, 0];
% heatmap = imagesc(grayscale_img);
% colormap(colormap_custom);
% 
% % Add a colorbar legend
% colorbar('Ticks', [0, 50, 100, 150, 200], 'TickLabels', {'0', '50', '100', '150', '200'});
% ylabel('Intensity');
% 
% % Add title
% title('Starry Night - Van Gogh Heatmap');
% 
% % Display the plot
% axis off; % Turn off axis labels and ticks
% axis image; % Maintain the aspect ratio

% Extract the red channel
red_channel = img(:,:,1);

% Create a heatmap using the red channel matrice
colormap_custom = jet; 
heatmap = imagesc(red_channel);

% Add a colorbar legend
colorbar;

% Add title
title('Starry Night - Van Gogh Heatmap');

% Display the plot
axis off; % Turn off axis labels and ticks
axis image; % Maintain the aspect ratio
